src/lib.rs

```diff
@@ -730,6 +548,146 @@ impl PartialEq for Color {
     }
 }
 
+//from HSLA to Color so you can do -> let new_color = Color::from(&some_hsla);
+impl From<&HSLA> for Color {
+    fn from(color: &HSLA) -> Self {
+        Color {
+            hue: Hue::from(color.h),
+            saturation: clamp(0.0, 1.0, color.s),
+            lightness: clamp(0.0, 1.0, color.l),
+            alpha: clamp(0.0, 1.0, color.alpha),
+        }
+    }
+}
+
+//from RGBA to Color so you can do -> let new_color = Color::from(&some_rgba);
+impl From<&RGBA<u8>> for Color {
+    fn from(color: &RGBA<u8>) -> Self {
+        let max_chroma = u8::max(u8::max(color.r, color.g), color.b);
+        let min_chroma = u8::min(u8::min(color.r, color.g), color.b);
+
+        let chroma = max_chroma - min_chroma;
+        let chroma_s = Scalar::from(chroma) / 255.0;
+
+        let r_s = Scalar::from(color.r) / 255.0;
+        let g_s = Scalar::from(color.g) / 255.0;
+        let b_s = Scalar::from(color.b) / 255.0;
+
+        let hue = 60.0
+            * (if chroma == 0 {
+                0.0
+            } else if color.r == max_chroma {
+                mod_positive((g_s - b_s) / chroma_s, 6.0)
+            } else if color.g == max_chroma {
+                (b_s - r_s) / chroma_s + 2.0
+            } else {
+                (r_s - g_s) / chroma_s + 4.0
+            });
+
+        let lightness = (Scalar::from(max_chroma) + Scalar::from(min_chroma)) / (255.0 * 2.0);
+        let saturation = if chroma == 0 {
+            0.0
+        } else {
+            chroma_s / (1.0 - Scalar::abs(2.0 * lightness - 1.0))
+        };
+        Self::from(&HSLA{ h: hue, s: saturation, l: lightness, alpha: color.alpha })
+    }
+}
+
+//from RGBA to Color so you can do -> let new_color = Color::from(&some_rgba);
```

Comment:
All of these comments don't really seem too helpful. It would be great if we could remove them.

Reply:
Yes, i'll do that now :)